src/Utils.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.0 <0.9.0;
+
+contract Utils {
+    address internal constant CONSOLE2_ADDRESS = address(0x000000000000000000636F6e736F6c652e6c6f67);
+    uint256 internal constant UINT256_MAX =
+        115792089237316195423570985008687907853269984665640564039457584007913129639935;
+
+    function bound(uint256 x, uint256 min, uint256 max) internal virtual returns (uint256 result) {
+        require(min <= max, "Test bound(uint256,uint256,uint256): Max is less than min.");
+
+        uint256 size = max - min;
+
+        if (size == 0) {
+            result = min;
+        } else if (size == UINT256_MAX) {
+            result = x;
+        } else {
+            ++size; // make `max` inclusive
+            uint256 mod = x % size;
+            result = min + mod;
+        }
+        
+        // Log the result with console2.log if supported
+        (bool status, ) = address(CONSOLE2_ADDRESS).staticcall(
+            abi.encodeWithSignature("log(string,uint)", "Bound result", result)
```

Comment:
This is the same mistake hardhat's `console.sol` made which is why `console2.sol` was needed 🙂

```suggestion
            abi.encodeWithSignature("log(string,uint256)", "Bound result", result)
```
Though, I'm not sure I fully understand why we wouldn't just import the console and call it, or declare `event log_named_uint` at the top of this file? There's lot of related issues, where's the best place to get an explanation of the new architecture and UX being implemented in this PR?

Reply:
Re redeclaring the `log_named_uint` event - we cannot do that because there would be namespace collisions with the events defined in `Assertions`.

But yeah, importing `console2.sol` should be fine. Thoughs @ZeroEkkusu?